src/cljc/matcher_combinators/matchers.cljc

```diff
@@ -114,6 +120,11 @@
   [transform-actual-fn expected]
   (core/->ViaMatcher transform-actual-fn expected))
 
+(defn either
+  "A matcher that successfully matches if one of the two provided matchers matches."
+  [left-expected right-expected]
+  (core/->Either left-expected right-expected))
```

Comment:
I wonder if `either` should take in an arbitrary number of arguments instead of just 2.

It starts to look like `clojure.core/some-fn`, which makes me wonder if we should also introduce an analog to `clojure.core/every-pred`, say `m/every`, so you can do things like

```clojure
(is 
  (match?
    (m/every (m/prefix [1 2 3]) #(> (count %) 5))
    (some-expr))
  "a list starting with 1 2 3 that has more than 5 elems")
```

Reply:
Looks a little bit more complex and may cause some confusion for users, I suggest sticking with either or even drop it for now, fine with keeping it though.

Reply:
We've got two separate problems here:
- apply a single matcher to every element in a sequence
- apply n matchers to a single element

`either` definitely leaves me wanting something like the `m/every` that you propose, so, for me, the pair of `seq-of` and `every` seem good from a functionality perspective. We should explore the names a bit more.

Reply:
How about `any` instead of `Either`? I'd prefer to avoid anything that implies monads :)

Reply:
I'm fine with `any`; `either` to me connotes two-ness ("either one or the other"). 
note that `any` has the downside of potentially being conflated with clojure's `(defn any? [x] true)` which is pretty different

Reply:
`any-of` and `all-of`?

Reply:
done in https://github.com/nubank/matcher-combinators/pull/191/commits/c38bd9c6611da087ba8b5c0f60fd455310924c63